webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,398 @@
+     * Updates provided answerTokens state on input change
+     */
+    textInputOnChange = (answerToken: string) => (v: string) => {
+        const answers = this.state.answers;
+        answers[answerToken] = v;
+
+        this.setState({ answers })
+    }
+
+    renderCore() {
+        const { visible, tsOutput, projectView, answers, currentQuestion, config } = this.state;
+
+        const actions: sui.ModalButton[] = [
+            {
+                label: lf("Back"),
+                onclick: this.backPage,
+                icon: 'arrow left',
+                className: 'arrow left',
+            },
+            {
+                label: lf("Next"),
+                onclick: this.nextPage,
+                icon: 'arrow right',
+                className: 'arrow right',
+            },
+            {
+                label: lf("Cancel"),
+                onclick: this.hide,
+                icon: "cancel",
+                className: "cancel lightgrey"
+            },
+            {
+                label: lf("Done"),
+                onclick: this.confirm,
+                icon: "check",
+                className: "approve positive"
+            }
+        ];
+
+        const currQ = config.questions[currentQuestion];
+
+        return (
+            <sui.Modal isOpen={visible} className="snippetBuilder" size="large"
+                closeOnEscape={false} closeIcon={false} closeOnDimmerClick={false} closeOnDocumentClick={false}
+                dimmer={true} buttons={actions} header={config.name}
+            >
+                <div>
+                    <div className="list">
+                        {currQ &&
+                            <div>
+                                <div>{currQ.title}</div>
+                                <div className='list horizontal'>
+                                    {currQ.inputs.map((input: any) =>
+                                        <div>
+                                            <sui.Input
+                                                label={input.label && input.label}
+                                                value={answers[input.answerToken] || ''}
+                                                onChange={this.textInputOnChange(input.answerToken)}
+                                            />
+                                        </div>
+                                    )}
+                                </div>
+                            </div>
+                        }
+                    </div>
+                    <div id="snippetBuilderOutput">
+                        {projectView && <md.MarkedContent markdown={this.generateOutputMarkdown(output)} parent={projectView} />}
+                    </div>
+                </div>
+            </sui.Modal>
+        )
+    }
+}
+
+// This will be passed down as a prop but is currently static
+const staticConfig: any = {
```

Comment:
use a function so that lf() calls a resolve late enough to pick up localization strings. This configuration should come from the target.